sklearn/ensemble/gradient_boosting.py

```diff
@@ -998,7 +998,10 @@ def fit(self, X, y, sample_weight=None, monitor=None):
 
         check_consistent_length(X, y, sample_weight)
 
-        y = self._validate_y(y)
+        if isinstance(self, GradientBoostingClassifier):
```

Comment:
This is now a bit silly. We have a method called _validate_y in both regressor and classifier to take advantage of polymorphism. Better off adding sample_weight parameter to both

Reply:
Actually I initially did this. But I kept remembering not to hinder with other classifier's or regressor's methods.

Reply:
Stylistic decisions, while affecting readability and maintainability, can be hard to weigh up.

As they say:

> There should be one-- and preferably only one --obvious way to do it.
> Although that way may not be obvious at first unless you're Dutch.

I'm part dutch :O)